sdk/file_sdk/UnitTests/pfile_ut/pfile_tests.cpp

```diff
@@ -0,0 +1,55 @@
+#include <QString>
+#include <QtTest>
+#include <string>
+#include <QTStreamImpl.h>
+#include <PFileFileFormat/pfile_file_format.h>
+
+using mip::file::PFileFileFormat;
+
+class pfile_tests : public QObject
+{
+  Q_OBJECT
+
+public:
+  pfile_tests();
+
+private Q_SLOTS:
+  void GetTags_FileWithTag_ReturnCorrectTag();
+  void GetTags_FileWithoutTag_ReturnZeroTags();
+};
+
+pfile_tests::pfile_tests()
+{
+}
+
+void pfile_tests::GetTags_FileWithTag_ReturnCorrectTag()
+{
+  foreach( const QString &fileName, QDir(":files/taged_pfiles").entryList() )
+  {
+    QFile file(":files/taged_pfiles/" + fileName);
+    QSharedPointer<QDataStream> obj = QSharedPointer<QDataStream>(new QDataStream(&file));
+    auto stream = QTStreamImpl::Create(obj);
+    PFileFileFormat pFileFileFormat(stream, ".ptxt");
+    auto tags = pFileFileFormat.GetTags();
+    QVERIFY2(tags.size() == 1, "Tags count shoud be 1");
+    mip::Tag tag = tags[0];
+    QVERIFY2(tag.GetLabelName() == "General", "Tag is different than expected");
```

Comment:
You should check all properties, not just the name

Reply:
I will use the "VerifyTags" function when it will be committed 

---
In reply to: [136312003](https://github.com/AzureAD/rms-sdk-for-cpp/pull/139#discussion_r136312003) [](ancestors = 136312003)